### PR TITLE
[POC] PIM-4526: Refactor copier to works with new media management

### PR DESCRIPTION
**Pull request by solivier**
| Q                  | A
| ------------------ | ---
| Migration script?  | n
| Behats             | running
| Specs & Static     | y
| Changelog updated  | y

### CODE REVIEW

File: src/Akeneo/Component/FileStorage/FileFactory.php [hunk 1]
```diff
@@ -78,6 +78,6 @@ public function createFromFile(FileInterface $file, $destFsAlias, $key = null)
         $newFile->setStorage($destFsAlias);
         $newFile->setKey($key);
 
-       return $newFile;
+        return $newFile;
```
Comment (jjanvier) [inline src/Akeneo/Component/FileStorage/FileFactory.php]: revert

File: src/Pim/Component/Catalog/Updater/Copier/MediaAttributeCopier.php [hunk 2]
```diff
@@ -106,57 +112,13 @@ protected function copySingleValue(
...
-     * @param ProductValueInterface $toValue
-     */
-    protected function deleteMedia(ProductValueInterface $toValue)
-    {
-        if (null !== $media = $toValue->getMedia()) {
-            $media->setOriginalFilename(null);
-            $media->setFilename(null);
-            $media->setMimeType(null);
+            $toValue->setMedia($file);
+            $file->setOriginalFilename($fromValue->getMedia()->getOriginalFilename());
```
Comment (jjanvier) [inline src/Pim/Component/Catalog/Updater/Copier/MediaAttributeCopier.php]: that works, but I would invert those 2 lines to increase understanding

File: src/Pim/Component/Catalog/Updater/Copier/MediaAttributeCopier.php [hunk 3]
```diff
@@ -19,31 +19,37 @@
...
+        RawFileFetcherInterface $rawFileFetcher,
+        RawFileStorerInterface $rawFileStorer,
+        MountManager $mountManager,
         array $supportedFromTypes,
         array $supportedToTypes
     ) {
         parent::__construct($productBuilder, $attrValidatorHelper);
-        $this->mediaManager       = $mediaManager;
-        $this->mediaFactory       = $mediaFactory;
+        $this->rawFileFetcher     = $rawFileFetcher;
```
Comment (jjanvier) [inline src/Pim/Component/Catalog/Updater/Copier/MediaAttributeCopier.php]: add those changes to the changelog please

File: src/Pim/Component/Catalog/Updater/Copier/MediaAttributeCopier.php [hunk 4]
```diff
@@ -106,57 +112,13 @@ protected function copySingleValue(
...
             }
 
-            $mediaHasFileName = false;
-            if (null !== $fromValue->getMedia()) {
-                $originalFileName = $fromValue->getMedia()->getOriginalFilename();
-                if (!empty($originalFileName)) {
-                    $mediaHasFileName = true;
-                }
-            }
+            $filesystem = $this->mountManager->getFilesystem('storage');
```
Comment (jjanvier) [inline src/Pim/Component/Catalog/Updater/Copier/MediaAttributeCopier.php]: add a todo to not hardcode storage please (we'll do all them at the end, dunno what's the good way to do it at the moment)

File: src/Akeneo/Component/FileStorage/FileFactory.php (hunk 1 above)
Comment (solivier) [inline src/Akeneo/Component/FileStorage/FileFactory.php]: why ? the return was not correctly aligned

**Comment by grena** [comment]
Looks GTM to me

**Pull request merged**